src/Cli/dotnet/Commands/Test/TestCommandDefinition.cs

```diff
@@ -0,0 +1,381 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.CommandLine;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.DotNet.Cli.CommandLine;
+using Microsoft.DotNet.Cli.Extensions;
+using Command = System.CommandLine.Command;
+
+namespace Microsoft.DotNet.Cli.Commands.Test;
+
+internal static class TestCommandDefinition
+{
+    public enum TestRunner
+    {
+        VSTest,
+        MicrosoftTestingPlatform
+    }
+
+    private sealed class GlobalJsonModel
+    {
+        [JsonPropertyName("test")]
+        public GlobalJsonTestNode Test { get; set; } = null!;
+    }
+
+    private sealed class GlobalJsonTestNode
+    {
+        [JsonPropertyName("runner")]
+        public string RunnerName { get; set; } = null!;
+    }
+
+    public const string Name = "test";
+    public static readonly string DocsLink = "https://aka.ms/dotnet-test";
+
+    public static readonly Option<string> SettingsOption = new Option<string>("--settings", "-s")
```

Comment:
Eventually I would love to separate all those options to different classes, one that's VSTest-specific, one that's MTP-specific, and one that's common.